### PR TITLE
report `CheckCode` for `check` failure

### PR DESCRIPTION
**Pull request by jmartin-tech**
`fail_with` should not be used in `check` method, convert to return the `Unknown` status when the endpoint required cannot be reached.

## Verification

List the steps needed to make sure this thing works

- [ ] See https://github.com/rapid7/metasploit-framework/pull/13847#issuecomment-661964193
- [ ] Provide an invalid target
- [ ] ***Verify*** failure is reported properly

### CODE REVIEW

File: modules/auxiliary/gather/peplink_bauth_sqli.rb [hunk 1]
```diff
@@ -196,7 +196,7 @@ def check
         'method' => 'GET',
         'cookie' => "bauth=' or #{payload}--"
       })
-      fail_with 'Unable to connect to target' unless res
+      fail_with 'Unable to connect to target' , "#{target_uri.path}" unless res
```
Comment (smcintyre-r7) [inline modules/auxiliary/gather/peplink_bauth_sqli.rb:199]: Shouldn't the first argument to fail_with be a `Failure::` constant? Something like `fail_with Failure::Unreachable, 'Unable to connect to target' unless res`.

**Review by smcintyre-r7 — CHANGES_REQUESTED** [review]
I was able to reproduce the original issue.

```
msf6 auxiliary(scanner/smb/smb_enum_gpp) > use auxiliary/gather/peplink_bauth_sqli 
msf6 auxiliary(gather/peplink_bauth_sqli) > show options 

Module options (auxiliary/gather/peplink_bauth_sqli):

   Name           Current Setting  Required  Description
   ----           ---------------  --------  -----------
   AdminOnly      false            yes       Only attempt to retrieve cookies of privilegied users (admins)
   BypassLogin    false            yes       Just bypass login without trying to leak the cookies of active sessions
   EnumPrivs      false            yes       Retrieve the privilege associated with each session
   EnumUsernames  false            yes       Retrieve the username associated with each session
   LimitTries                      no        The max number of sessions to try (from most recent), set to avoid checking expired ones needlessly
   Proxies                         no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS                          yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT          80               yes       The target port (TCP)
   SSL            false            no        Negotiate SSL/TLS for outgoing connections
   TARGETURI      /                yes       The target URI
   VHOST                           no        HTTP server virtual host

msf6 auxiliary(gather/peplink_bauth_sqli) > set RHOSTS 127.0.0.1
RHOSTS => 127.0.0.1
msf6 auxiliary(gather/peplink_bauth_sqli) > run
[*] Running module against 127.0.0.1

[-] Auxiliary aborted due to failure: Unable to connect to target: 
[*] Auxiliary module execution completed
msf6 auxiliary(gather/peplink_bauth_sqli) >
```

File: modules/auxiliary/gather/peplink_bauth_sqli.rb (hunk 1 above)
Comment (jmartin-r7) [reply modules/auxiliary/gather/peplink_bauth_sqli.rb:199]: Will update, I think I might expand the check to be a bit more explicit and do a bit more pre-validation as well.

**Review by smcintyre-r7 — APPROVED** [review]
Looks alot better now, I tried both running the module and calling the check method.

```
msf6 auxiliary(gather/peplink_bauth_sqli) > run
[*] Running module against 127.0.0.1

[-] Target does not seem to be vulnerable
[*] Auxiliary module execution completed
msf6 auxiliary(gather/peplink_bauth_sqli) > check
[*] 127.0.0.1:80 - Cannot reliably check exploitability. Unable to connect to /
msf6 auxiliary(gather/peplink_bauth_sqli) >
```

I'll have this merged in momentarily.

**Pull request merged**